src/overlays/actors/ovl_Demo_Effect/z_demo_effect.c

```diff
@@ -32,33 +37,284 @@ const ActorInit Demo_Effect_InitVars = {
+
+        switch (params) {
+            case 0:
+                temp_f0 *= 1.2f;
+                break;
+
+            case 1:
+                temp_f0 *= 0.6f;
+                break;
+
+            case 2:
+                temp_f0 *= 1.2f;
+                break;
+
+            case 3:
+                temp_f0 *= 2.0f;
+                break;
+        }
+
+        this->actor.scale.x = temp_f0;
+        this->actor.scale.z = temp_f0;
+        func_808CDAD0(temp_f12);
+        func_800B8FE8(&this->actor, NA_SE_EV_TIMETRIP_LIGHT - SFX_FLAG);
+    } else {
+        func_808CDAD0(1.0f);
+        Actor_MarkForDeath(&this->actor);
+    }
+}
+
+void func_808CDCEC(DemoEffect* this, GlobalContext* globalCtx) {
+    func_800B8FE8(&this->actor, NA_SE_EV_TIMETRIP_LIGHT - SFX_FLAG);
+
+    if (SkelCurve_Update(globalCtx, &this->skelCurve)) {
+        SkelCurve_SetAnim(&this->skelCurve, &object_efc_tw_CurveAnim_000050, 1.0f, 60.0f, 59.0f, 0.0f);
+        this->actionFunc = func_808CDBDC;
+        this->unk_16C = 0;
+    }
+}
+
+void func_808CDD70(DemoEffect* this, GlobalContext* globalCtx) {
+    Actor_SetScale(&this->actor, this->actor.scale.x - 0.02f);
+
+    this->unk_16C++;
+    if (this->actor.scale.x < 0.02f) {
+        Actor_MarkForDeath(&this->actor);
+    }
+}
+
+void func_808CDDE0(DemoEffect* this, GlobalContext* globalCtx) {
+    Actor_SetScale(&this->actor, (this->actor.scale.x * 0.5f) + 0.2f);
+
+    this->unk_16C++;
+    if (this->unk_16C >= 3) {
+        this->actionFunc = func_808CDD70;
+    }
+}
+
+void DemoEffect_Update(Actor* thisx, GlobalContext* globalCtx) {
+    DemoEffect* this = THIS;
+
+    this->actionFunc(this, globalCtx);
+}
+
+s32 func_808CDE78(GlobalContext* globalCtx, SkelAnimeCurve* skelCuve, s32 limbIndex, Actor* thisx) {
+    s32 pad;
+    DemoEffect* this = THIS;
+    u32 sp4C = globalCtx->gameplayFrames;
+
+    OPEN_DISPS(globalCtx->state.gfxCtx);
+
+    func_8012C2DC(globalCtx->state.gfxCtx);
+
+    gDPSetPrimColor(POLY_XLU_DISP++, 0, 0x80, 170, 255, 255, 255);
+    gDPSetEnvColor(POLY_XLU_DISP++, this->unk_168[0], this->unk_168[1], this->unk_168[2], 255);
+    gSPSegment(POLY_XLU_DISP++, 0x08,
+               Gfx_TwoTexScroll(globalCtx->state.gfxCtx, 0, (sp4C * 6) % 1024, 255 - ((sp4C * 16) % 256), 0x100, 0x40,
+                                1, (sp4C * 4) % 512, 127 - ((sp4C * 12) % 128), 0x80, 0x20));
+
+    CLOSE_DISPS(globalCtx->state.gfxCtx);
+
+    if (limbIndex == 0) {
+        LimbTransform* transform = skelCuve->transforms;
+
+        transform->scale.x = 0x400;
+        transform->scale.y = 0x400;
+        transform->scale.z = transform->scale.x;
+    }
+
+    return true;
+}
+
+void func_808CDFF8(Actor* thisx, GlobalContext* globalCtx) {
+    OPEN_DISPS(globalCtx->state.gfxCtx);
+
+    POLY_XLU_DISP = Gfx_CallSetupDL(POLY_XLU_DISP, 25);
+
+    Matrix_Scale(2.0f, 2.0f, 2.0f, MTXMODE_APPLY);
+    SkelCurve_Draw(thisx, globalCtx, &THIS->skelCurve, func_808CDE78, NULL, 1, thisx);
```

Comment:
```suggestion
    SkelCurve_Draw(thisx, globalCtx, &THIS->skelCurve, func_808CDE78, NULL, 1, thisx);
```
Not enough room on the stack for this cast, or is something push too far down?

Reply:
Pushed down by 4: https://decomp.me/scratch/ot2Fv

Reply:
https://decomp.me/scratch/TUU8D, seems to work with a `gfxCtx` temp which I think is preferable.